docs/rfc/block-registration.md

```diff
@@ -0,0 +1,369 @@
+This RFC is intended to serve both as a specification and as documentation for the implementation of runtime-agnostic block type registration.
```

Comment:
Some important aspects here:

 - The block library package are served as an npm package, we don't know where these will be run
 - The block repository (one of the 9 projects) need to be able to get block information outside of WordPress

I guess this could be added somewhere